fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/spring/ModeShapeRepositoryFactoryBean.java

```diff
@@ -60,21 +61,24 @@
      * @throws Exception
      */
     @PostConstruct
-    public void buildRepository() throws Exception {
-        LOGGER.info("Using repo config (classpath): {}", repositoryConfiguration.getURL());
-
-        getPropertiesLoader().loadSystemProperties();
-
-        final RepositoryConfiguration config =
-                RepositoryConfiguration.read(repositoryConfiguration.getURL());
-        repository = modeShapeEngine.deploy(config);
-
-        // next line ensures that repository starts before the factory is used.
-        final org.modeshape.common.collection.Problems problems =
-                repository.getStartupProblems();
-        for (final org.modeshape.common.collection.Problem p : problems) {
-            LOGGER.error("ModeShape Start Problem: {}", p.getMessageString());
-            // TODO determine problems that should be runtime errors
+    public void buildRepository() {
+        try {
+            LOGGER.info("Using repo config (classpath): {}", repositoryConfiguration.getURL());
+            getPropertiesLoader().loadSystemProperties();
+
+            final RepositoryConfiguration config =
+                    RepositoryConfiguration.read(repositoryConfiguration.getURL());
+            repository = modeShapeEngine.deploy(config);
+
+            // next line ensures that repository starts before the factory is used.
+            final org.modeshape.common.collection.Problems problems =
+                    repository.getStartupProblems();
+            for (final org.modeshape.common.collection.Problem p : problems) {
+                LOGGER.error("ModeShape Start Problem: {}", p.getMessageString());
+                // TODO determine problems that should be runtime errors
+            }
+        } catch (Exception e) {
+            throw new RepositoryRuntimeException(e);
```

Comment:
I may be being really too picky here, but to my understanding, this is not a `RepositoryRuntimeException` precisely because the repository is not running. We _could_ put in a new `RepositoryFailedStartup` kind of thing, or maybe MODE already has such thing, or maybe I'm just being too picky...